static/app/components/notificationActions/notificationActionItem.tsx

```diff
@@ -0,0 +1,344 @@
+            : t('service')}
+        </div>
+      </Fragment>
+    );
+  }
+
+  async function handleDelete() {
+    const endpoint = `/organizations/${organization.slug}/notifications/actions/${action.id}/`;
+    try {
+      await api.requestPromise(endpoint, {
+        method: 'DELETE',
+      });
+      addSuccessMessage(t('Successfully deleted notification action'));
+      onDelete(index);
+    } catch (err) {
+      addErrorMessage(t('Unable to delete notification action'));
+    }
+  }
+
+  function handleCancel() {
+    if (action.id) {
+      setIsEditing(false);
+      return;
+    }
+    // Delete the unsaved notification action
+    onDelete(index);
+  }
+
+  async function handleSave() {
+    const formProps = getFormProps();
+    addLoadingMessage();
+    // TODO(enterprise): use "requires" to get data to send
+    // This is currently optimized for spike protection
+    const data = cloneDeep(updatedAction);
+
+    // Remove keys from the data if they are falsy
+    Object.keys(data).forEach(key => {
+      if (!data[key]) {
+        delete data[key];
+      }
+    });
+
+    try {
+      const resp = await api.requestPromise(formProps.apiEndpoint, {
+        method: formProps.apiMethod,
+        data: {...data, projects: [project.slug]},
+      });
+      addSuccessMessage(formProps.successMessage);
+      onUpdate(index, resp);
+      setUpdatedAction(resp);
+      setIsEditing(false);
+    } catch (err) {
+      addErrorMessage(formProps.errorMessage);
+    }
+  }
+
+  function handleChange(name: string, value: any) {
+    const modifiedAction = cloneDeep(updatedAction);
+    modifiedAction[name] = value;
+    setUpdatedAction(modifiedAction);
+  }
+
+  // Edit button is located outside of the form
+  function renderEditButton() {
+    const menuItems: MenuItemProps[] = [
+      {
+        key: 'notificationaction-delete',
+        label: t('Delete'),
+        priority: 'danger',
+        onAction: () => {
+          openConfirmModal({
+            message: t('Are you sure you want to delete this notification action?'),
+            onConfirm: handleDelete,
+          });
+        },
+      },
+    ];
+
+    // No edit mode for Sentry notifications
+    if (serviceType !== NotificationActionService.SENTRY_NOTIFICATION) {
+      menuItems.unshift({
+        key: 'notificationaction-edit',
+        label: t('Edit'),
+        onAction: () => setIsEditing(true),
+      });
+    }
+
+    return (
+      <DropdownMenu
+        items={menuItems}
+        trigger={triggerProps => (
+          <Button
+            {...triggerProps}
+            aria-label={t('Actions')}
+            size="xs"
+            icon={<IconEllipsis direction="down" size="sm" />}
+            onClick={e => {
+              e.stopPropagation();
+              e.preventDefault();
+
+              triggerProps.onClick?.(e);
+            }}
+          />
+        )}
+      />
+    );
+  }
+
+  function getFormProps() {
+    if (updatedAction.id) {
```

Comment:
if `updatedAction` exists, will it always have an id?
Should this just be checking the existence of updatedAction rather than `id`? otherwise if its undefined, this will throw an error right?

Reply:
`updatedAction` may not have an id if it hasn't been saved yet